presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -439,7 +439,7 @@ public static NullableValue parsePartitionValue(String partitionName, String val
                 return NullableValue.of(DOUBLE, parseDouble(value));
             }
 
-            if (HIVE_STRING.equals(hiveType) || hiveType.toString().toLowerCase().startsWith("varchar")) {
+            if (HIVE_STRING.equals(hiveType) || hiveType.getTypeSignature().getBase().equals(VARCHAR.toString())) {
```

Comment:
This won't work with parametrized VARCHAR. Use `hiveType.getTypeSignature().getBase().equals(VARCHAR.getTypeSignature().getBase())` instead.